src/arrayMethodReduce.js

```diff
@@ -1,12 +1,27 @@
-'use strict';
+"use strict";
 
 /**
  * Implement method Reduce
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let total = initialValue;
+    let startIndex = 0;
+
+    if (!this.length && !initialValue) {
+      throw new TypeError('Empty array and no initialValue');
+    }
+
+    if (arguments.length < 2) {
+      total = this[0];
+      startIndex = 1;
+    }
+
+    for (let i = startIndex; i < this.length; i++) {
```

Comment:
```suggestion
    for (startIndex; i < this.length; i++) {
```

Reply:
i tried to use this syntax but got "i is not defined" error so i tried to apply it in another way